### ISSUE TITLE
feature request: hidden queues

### ISSUE THREAD

**Issue by ryanday36** [flux-framework/flux-core#5030]
There are scenarios where it can be useful to have a queue of nodes that is hidden from users who don't have access to run on that queue. Would it be possible to either add a 'hidden' property to queues that would tell user commands like `flux resource list` not to display them by default, or to have those user commands only display the queues that a user has access to by default?

**Comment by trws** [comment]
It occurred to me just now that having it be only queues that are usable by default would actually be consistent with how flux jobs or similar behave, where we only display the user's info by default, and show everything only with options. Not commenting on priority, but viewed in light of "by default only display relevant info" it seems more reasonable than having a magic "hide this thing" option somehow.

**Comment by grondo** [comment]
That's is a good point, however the queue policy may be configured and enforced by a plugin external to flux-core (in this case the flux-accounting priority plugin), and I'm unsure how we'd actually gather the usable queues from the flux-core tools.

**Comment by trws** [comment]
Makes sense. Hopefully we’ll have a better idea once #5027 gets shaken out.
